lib/purescript-cst/src/Language/PureScript/CST/Convert.hs

```diff
@@ -530,6 +531,55 @@ convertDeclaration fileName decl = case decl of
   ann =
     uncurry (sourceAnnCommented fileName) $ declRange decl
 
+  startSourcePos :: SourceToken -> Pos.SourcePos
+  startSourcePos = sourcePos . srcStart . tokRange . tokAnn
+
+  mkPartialInstanceName :: Maybe (Name Ident, SourceToken) -> QualifiedName (N.ProperName 'N.ClassName) -> [Type a] -> Either Text.Text N.Ident
+  mkPartialInstanceName nameSep cls args =
+    maybe (Left genName) (Right . ident . nameValue . fst) nameSep
+    where
+      -- instance name truncation will occur in desugaring process
+      -- when the final instance name is determined
+      genName :: Text.Text
+      genName = "$" <> className <> "$$" <> typeArgs
+
+      className :: Text.Text
+      className = do
+        let modName = (<> "_") . Text.replace "." "_" . N.runModuleName
+        (maybe "" modName (qualModule cls)) <> (N.runProperName $ qualName cls)
+
+      typeArgs :: Text.Text
+      typeArgs = Text.intercalate "$" $ fmap argName args
+
+      argName :: Type a -> Text.Text
+      argName = \case
+        -- These are only useful to disambiguate between overlapping instances
+        -- but they’re disallowed outside of instance chains. Since we’re
+        -- avoiding name collisions with unique identifiers anyway,
+        -- we don't need to render this constructor.
+        TypeVar{} -> ""
+        TypeConstructor _ qn -> N.runProperName $ qualName qn
+        TypeOpName _ qn -> N.runOpName $ qualName qn
+        TypeString _ _ ps -> prettyPrintStringJS ps
+
+        -- Typed holes are disallowed in instance heads
+        TypeHole{} -> ""
+        TypeParens _ t -> argName $ wrpValue t
+        TypeKinded _ t1 _ t2 -> argName t1 <> "$" <> argName t2
+        TypeRecord _ _ -> "Record"
+        TypeRow _ _ -> "Row"
+        TypeArrName _ _ -> "Function"
+        TypeWildcard{} -> "_"
+
+        -- Polytypes are disallowed in instance heads
+        TypeForall{} -> ""
+        TypeApp _ t1 t2 -> argName t1 <> "$" <> argName t2
+        TypeOp _ t1 op t2 ->
+          argName t1 <> "$" <> (N.runOpName $ qualName op) <> "$" <> argName t2
+        TypeArr _ t1 _ t2 -> argName t1 <> "_Arrow_" <> argName t2
+        TypeConstrained{} -> ""
+        TypeUnaryRow{} -> "EmptyRow"
```

Comment:
Is this right? Or should this be rendered as `""`. It looks like the only usage of this constructor is when [it's used with the deprecated `#` syntax](https://github.com/purescript/purescript/blob/master/lib/purescript-cst/src/Language/PureScript/CST/Parser.y#L312

Reply:
`Row Type` is rendered as `Row_Type`, so I think `# Type` should render to the same string:

```suggestion
        TypeUnaryRow{} -> "Row"
```